src/System.Reflection.DispatchProxy/tests/DispatchProxyTests.cs

```diff
@@ -386,5 +386,94 @@ public static void Invoke_Property_Setter_And_Getter_Invokes_Correct_Methods()
 
             Assert.Null(actualValue);
         }
+
+
+        [Fact]
+        public static void Proxy_Declares_Interface_Properties()
+        {
+            TestType_IPropertyService proxy = DispatchProxy.Create<TestType_IPropertyService, TestDispatchProxy>();
+            PropertyInfo propertyInfo = proxy.GetType().GetTypeInfo().GetDeclaredProperty("ReadWrite");
+            Assert.NotNull(propertyInfo);
+        }
+
```

Comment:
An Invoke_Correct_Methods() test for non-indexed properties would be appropriate here.